keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -117,59 +116,44 @@ serves the resourceVersion="0" list requests from reflectors today.
 
 Guard this by a `WatchCacheConsistentReads` feature gate.
 
-#### Use WithProgressNotify to enable automatic watch updates
+This alternative requires using `WatchProgressRequest` which is only available in etcd 3.4+, and so would
+require we make the kube-apiserver aware of etcd's minor version, which is described in more detail later.
 
-Create etcd watches with `WithProgressNotify` enabled (available in all etcd 3.x versions).
+<<[UNRESOLVED @serathius]>>
+Propose a method to make kube-apiserver aware of etcd version:
+* Flag providing etcd version to kube-apiserver `--storage-backend=etcd3.4`
+* Flag to enable/disable the feature `--allow-using-progress-notify`
+* Have kube-apiserver check cluster version in etcd `/version` endpoint. 
+  Retry the check logic if `WatchProgressRequest` fails.
+<<[/UNRESOLVED]>>
 
-When `WithProgressNotify` is enabled on an etcd watch, etcd sends progress
-events to the watch automatically. By default etcd sends progress events every
-10 minutes, which is not frequent enough to be useful for our needs, so we will
-modify etcd to send them more frequently.
+#### Use RequestProgress to enable automatic watch updates
 
-When an consistent LIST request is received and the watch cache is enabled:
+When a consistent LIST request is received and the watch cache is enabled:
 
-- Get the current revision from etcd for the resource type being served. The returned revision is strongly consistent (guaranteed to be the latest revision via a quorum read).
+- Get the current revision from etcd for the resource type being served. Use 
```

Comment:
We already added this  logic as part of watch-list KEP:
https://github.com/kubernetes/kubernetes/blob/3f247e59edfd4083242ad7271d076a38291760ff/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1246

However, it's slightly different - if you believe we can improve it's performance, please go ahead and change that.

Reply:
Awesome